src/main/java/codeu/controller/ConversationServlet.java

```diff
@@ -100,23 +102,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     }
 
     String conversationTitle = request.getParameter("conversationTitle");
-    if (!conversationTitle.matches("[\\w*]*")) {
-      request.setAttribute("error", "Please enter only letters and numbers.");
-      request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
-      return;
-    }
-
-    if (conversationStore.isTitleTaken(conversationTitle)) {
-      // conversation title is already taken, just go into that conversation instead of creating a
-      // new one
+    String groupTitle = request.getParameter("groupTitle");
+    if(conversationTitle != null){
+      if (!conversationTitle.matches("[\\w*]*")) {
+        request.setAttribute("error", "Please enter only letters and numbers.");
+        request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
+        return;
+      }
+
+      if (conversationStore.isTitleTaken(conversationTitle)) {
+        // conversation title is already taken, just go into that conversation instead of creating a
+        // new one
+        response.sendRedirect("/chat/" + conversationTitle);
+        return;
+      }
+
+      Conversation conversation =
+          new Conversation(UUID.randomUUID(), user.getId(), conversationTitle, Instant.now());
+
+      conversationStore.addConversation(conversation);
       response.sendRedirect("/chat/" + conversationTitle);
-      return;
     }
-
-    Conversation conversation =
-        new Conversation(UUID.randomUUID(), user.getId(), conversationTitle, Instant.now());
-
-    conversationStore.addConversation(conversation);
-    response.sendRedirect("/chat/" + conversationTitle);
+    else{
+      if (!groupTitle.matches("[\\w*]*")) {
+        request.setAttribute("error", "Please enter only letters and numbers.");
+        request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
+        return;
+      }
+
+      if (conversationStore.isTitleTaken(groupTitle)) {
+        // conversation title is already taken, just go into that conversation instead of creating a
+        // new one
+        response.sendRedirect("/chat/" + groupTitle);
+        return;
+      }
+
+      Conversation groupConversation =
+          new Conversation(UUID.randomUUID(), user.getId(), groupTitle, Instant.now(), true);
+
+      conversationStore.addConversation(groupConversation);
+      response.sendRedirect("/chat/" + groupTitle);
+    }
```

Comment:
I don't think you need all of this repeated code. Instead, you code do something like:
```java
// Ensures either a conversationTitle or a groupTitle exists, both not both; essentially XOR
if ((conversationTitle == null) == (groupTitle == null)) {
    throw new RuntimeException("Must have either conversationTitle OR groupTitle");
}

boolean isGroupTitle = conversationTitle == null;
String title = isGroupTitle? groupTitle : conversationTitle;

//Then all of the code you've repeated above using isGroupTitle and title instead.
```